start.py

```diff
@@ -536,40 +543,57 @@ def _command_condition(deployment, command_id, command_description):
             else:
                 parcel.deactivate()
+                parcel_found_state = 'is_deactivated'
+            # Only one parcel can be activated at a time, so once one is found, our work is done.
+            break
+    return parcel_found_state
 
-                for config in deployment.get_cm_config():
-                    if config['name'] == 'REMOTE_PARCEL_REPO_URLS':
-                        break
-                else:
-                    raise Exception('Failed to find remote parcel repo URLs configuration.')
-                parcel_repo_urls = config['value']
-
-                kafka_parcel_repo_url = '{}/{}'.format(KAFKA_PARCEL_REPO_URL, kafka_version)
-                logger.debug('Adding Kafka parcel repo URL (%s) ...', kafka_parcel_repo_url)
-                deployment.update_cm_config(
-                    {'REMOTE_PARCEL_REPO_URLS': '{},{}'.format(parcel_repo_urls,
-                                                               kafka_parcel_repo_url)}
-                )
 
-                logger.debug('Refreshing parcel repos ...')
-                deployment.refresh_parcel_repos()
-                kafka_parcel = next(parcel
-                                    for parcel in deployment.cluster(DEFAULT_CLUSTER_NAME).parcels
-                                    if parcel.product == 'KAFKA'
-                                    and parcel.version.split('-')[0] == kafka_version)
-                kafka_parcel.download().distribute().activate()
-            # Only one parcel can be activated at a time, so once one is found, our work is done.
+def _install_service_from_remote_repo(deployment, product, version):
+    for config in deployment.get_cm_config():
+        if config['name'] == 'REMOTE_PARCEL_REPO_URLS':
             break
+    else:
+        raise Exception('Failed to find remote parcel repo URLs configuration.')
+    parcel_repo_urls = config['value']
+
+    service_parcel_repo_url = '{}/{}'.format(SERVICE_PARCEL_REPO_URLS[product],
+                                             KUDU_PARCEL_VERSIONS[version] if product == 'KUDU' else version)
+    logger.debug('Adding parcel repo URL (%s) ...', service_parcel_repo_url)
+    deployment.update_cm_config(
+        {'REMOTE_PARCEL_REPO_URLS': '{},{}'.format(parcel_repo_urls,
+                                                   service_parcel_repo_url)}
+    )
+
+    logger.debug('Refreshing parcel repos ...')
+    deployment.refresh_parcel_repos()
+    service_parcel = next(parcel
+                          for parcel in deployment.cluster(DEFAULT_CLUSTER_NAME).parcels
+                          if parcel.product == product
+                          and parcel.version.split('-')[0] == version)
+    service_parcel.download().distribute().activate()
+
+
+def _configure_kafka(deployment, cluster, kafka_version, cdh_version):
+    cdh_version_tuple = tuple([int(i) for i in cdh_version.split('.')])
+    cdh_5_13_0_tuple = 5, 13, 0
```

Comment:
There's no dependency on CDH versions with Kafka. We just need to change the logic to handle the different possible scenarios (e.g. Kafka installed or not installed in the Docker images for the topology)

Reply:
Done.